lib/path.dart

```diff
@@ -281,9 +281,25 @@ String joinAll(Iterable<String> parts) => context.joinAll(parts);
 ///       // -> ['http://dartlang.org', 'path', 'to', 'foo']
 List<String> split(String path) => context.split(path);
 
+/// Canonicalizes [path].
+///
+/// This is guaranteed to return the same path for two different input paths
```

Comment:
How about "input paths" -> "input strings" to be clearer that it's not talking about things from the file system's perspective?